cpp/src/plasma/malloc.cc

```diff
@@ -154,14 +138,6 @@ void* fake_mmap(size_t size) {
     ARROW_LOG(ERROR) << "mmap failed with error : " << std::strerror(errno);
     return pointer;
   }
-  // Attempt to mlock the mmaped region of memory (best effort).
-  int rv = mlock(pointer, size);
-  if (rv != 0) {
-    ARROW_LOG(WARNING) << "mlock failed with error : " << std::strerror(errno);
-  }
-  ARROW_LOG(INFO) << "mlocking pointer " << pointer << " size " << size
-                     << " success " << rv;
-  memset(pointer, 0xff, size);
```

Comment:
By getting rid of this memset, writing in my benchmarks takes 2x to 3x as long in the huge page case (that's the only performance difference I can measure from this commit); if there is no other way to achieve the same effect we will need to bring it back.